src/datadog/w3c_propagation.cpp

```diff
@@ -0,0 +1,396 @@
+#include "w3c_propagation.h"
+
+#include <algorithm>
+#include <cassert>
+#include <cstddef>
+#include <regex>
+#include <utility>
+
+#include "dict_reader.h"
+#include "hex.h"
+#include "parse_util.h"
+#include "tags.h"
+
+namespace datadog {
+namespace tracing {
+namespace {
+
+// Return a predicate that returns whether its `char` argument is any of the
+// following:
+//
+// - outside of the ASCII inclusive range `[lowest_ascii, highest_ascii]`
+// - equal to one of the `disallowed_characters`.
+//
+// `verboten` is used with `std::replace_if` to sanitize field values within the
+// tracestate header.
```

Comment:
The entire header or just our part of it?

Reply:
Just our part of it.